dask_sql/physical/rel/logical/table_scan.py

```diff
@@ -48,11 +51,15 @@ def convert(
         df = dc.df
         cc = dc.column_container
 
-        # Make sure we only return the requested columns
-        row_type = table.getRowType()
-        field_specifications = [str(f) for f in row_type.getFieldNames()]
-        cc = cc.limit_to(field_specifications)
+        # If the 'TableScan' instance contains projected columns only retrieve those columns
+        # otherwise get all projected columns from the 'Projection' instance, which is contained
+        # in the 'RelDataType' instance, aka 'row_type'
+        if table_scan.containsProjections():
+            field_specifications = table_scan.getTableScanProjects()
+        else:
+            field_specifications = [str(f) for f in table.getRowType().getFieldNames()]
```

Comment:
```suggestion
            field_specifications = table.getRowType().getFieldNames()
```

Do we have to loop over and cast field names with `str`?